internal/services/oracle/oracle_resource_anchor_resource.go

```diff
@@ -0,0 +1,196 @@
+	return "azurerm_oracle_resource_anchor"
+}
+
+func (r ResourceAnchorResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient09.ResourceAnchors
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ResourceAnchorResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			id := resourceanchors.NewResourceAnchorID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			param := resourceanchors.ResourceAnchor{
+				Location:   model.Location,
+				Tags:       pointer.To(model.Tags),
+				Properties: &resourceanchors.ResourceAnchorProperties{},
+			}
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (ResourceAnchorResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient09.ResourceAnchors
+			id, err := resourceanchors.ParseResourceAnchorID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state := ResourceAnchorResourceModel{
+				Name:              id.ResourceAnchorName,
+				ResourceGroupName: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = model.Location
+				state.Tags = pointer.From(model.Tags)
+
+				if props := model.Properties; props != nil {
+					state.LinkedCompartmentID = pointer.From(props.LinkedCompartmentId)
+					state.ProvisioningState = string(pointer.From(props.ProvisioningState))
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (ResourceAnchorResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient09.ResourceAnchors
+			id, err := resourceanchors.ParseResourceAnchorID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model ResourceAnchorResourceModel
+			if err = metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding err: %+v", err)
+			}
+
+			update := resourceanchors.ResourceAnchorUpdate{}
+
```

Comment:
Presumably missing `linked_compartment_id`? Otherwise, `linked_compartment_id` shall be marked `ForceNew`.